core/dbt/parser/base.py

```diff
@@ -280,9 +279,18 @@ def update_parsed_node_config_dict(
     def update_parsed_node_relation_names(
         self, parsed_node: IntermediateNode, config_dict: Dict[str, Any]
     ) -> None:
-        self._update_node_database(parsed_node, config_dict)
-        self._update_node_schema(parsed_node, config_dict)
-        self._update_node_alias(parsed_node, config_dict)
+
+        if parsed_node.resource_type == "snapshot":
+            if "target_database" in config_dict and config_dict["target_database"]:
+                parsed_node.database = config_dict["target_database"]
+            if "target_schema" in config_dict and config_dict["target_schema"]:
+                parsed_node.schema = config_dict["target_schema"]
```

Comment:
For my own understanding: Previously we were calling also `self._update_node_[database|schema|alias]` on snapshots nodes. Were those no-ops and are no longer necessary?